src/boot/efi/boot.c

```diff
@@ -2405,7 +2405,11 @@ static EFI_STATUS image_start(
                 loaded_image->LoadOptionsSize = strsize16(options);
 
                 /* Try to log any options to the TPM, especially to catch manually edited options */
-                (void) tpm_log_load_options(options, NULL);
+                (void) tpm_log_load_options(
+                                loaded_image->LoadOptions,
+                                loaded_image->LoadOptionsSize,
+                                u"systemd-boot entry load options",
```

Comment:
please don't. this makes the efi tpm measurement log a lot less useful, as this means it's not going to be self-explanatory anymore. i.e. we want that people can reason over the efi tpm measurement log and validate that the PCRs are in the way they are expected to be. But this only works if we have a rough idea what was measured. Now, given that the cmdline can be specifie don the fly by a human user at the keyboard we really should include the full string in the tpm event log entry description field. (it's ok to prefix it with something friendly, but the string should be there)

Reply:
or to say this differently: if you call the function as you do there, the first param is simply hashed, and that hash is available in the efi tpm logs, but not the actual data.